pipes/rules/metagenomics.rules

```diff
@@ -74,65 +75,104 @@ rule diamond:
-        kraken_db   = objectify_remote(expand("{path_prefix}/{krakenfile}", path_prefix=config["kraken_db"], krakenfile=krakenfiles)),
-        taxonomy_db = objectify_remote(expand("{path_prefix}/taxonomy/{taxfile}", path_prefix=config["kraken_db"], taxfile=taxfiles_kraken))
-    output: 
-        report = os.path.join(config["data_dir"], config["subdirs"]["metagenomics"], "{sample}.{adjective,raw|cleaned}.kraken.report"),
-        reads  = os.path.join(config["data_dir"], config["subdirs"]["metagenomics"], "{sample}.{adjective,raw|cleaned}.kraken.reads.gz")
-    resources: 
-        threads = int(config.get("number_of_threads", 1)),
-        mem     = 120
-    params: 
-        UGER = config.get('UGER_queues', {}).get('long', '-l h_rt=36:00:00')
-    run:
-        kraken_db_prefix = strip_protocol(config["kraken_db"], relative=True)
-        shell("{config[bin_dir]}/metagenomics.py kraken {input.bam} "+kraken_db_prefix+" --outReads {output.reads} --outReport {output.report} --threads {resources.threads}")
-
 rule align_rna:
     input:
         bam         = os.path.join(config["data_dir"], config["subdirs"]["per_sample"], "{sample}.{adjective}.bam"),
         rna_bwa_db  = objectify_remote(expand("{path_prefix}.{ext}", path_prefix=config["align_rna_db"], ext=rna_bwa_ext)),
         taxonomy_db = objectify_remote(expand("{path_prefix}/{taxfile}", path_prefix=config["taxonomy_db"], taxfile=taxfiles))
-    output: 
+    output:
         report         = os.path.join(config["data_dir"], config["subdirs"]["metagenomics"], "{sample}.{adjective,raw|cleaned}.rna_bwa.report"),
         nodupes_report = os.path.join(config["data_dir"], config["subdirs"]["metagenomics"], "{sample}.{adjective,raw|cleaned}.rna_bwa.nodupes.report"),
         bam            = os.path.join(config["data_dir"], config["subdirs"]["metagenomics"], "{sample}.{adjective,raw|cleaned}.rna_bwa.bam"),
         lca            = os.path.join(config["data_dir"], config["subdirs"]["metagenomics"], "{sample}.{adjective,raw|cleaned}.rna_bwa.lca.gz"),
         nodupes_lca    = os.path.join(config["data_dir"], config["subdirs"]["metagenomics"], "{sample}.{adjective,raw|cleaned}.rna_bwa.lca_nodupes.gz")
-    resources: 
+    resources:
         threads = int(config.get("number_of_threads", 1)),
         mem     = 36
-    params: 
+    params:
         UGER = config.get('UGER_queues', {}).get('long', '-l h_rt=36:00:00')
     run:
         rna_bwa_path_prefix = strip_protocol(config["align_rna_db"], relative=True)
         taxonomy_db_prefix = strip_protocol(config["taxonomy_db"], relative=True)
         shell("{config[bin_dir]}/metagenomics.py align_rna {input.bam} "+rna_bwa_path_prefix+" "+taxonomy_db_prefix+" {output.nodupes_report} --dupeReport {output.report} --outBam {output.bam} --outReads {output.nodupes_lca} --dupeReads {output.lca} --JVMmemory {resources.mem}g --threads {resources.threads}")
 
+if config.get('kraken_execution', 'single') == 'multiple':
```

Comment:
What happens if config['kraken_execution'] is set to 'multiple' but the target is `.cleaned.kraken.report`? I'm not sure whether a rule can generate it in that case. Also, you might want to put 'kraken_execution' into config.yaml so people don't forget about it.

Reply:
Yeah I need to add that part.